lib/uklibparam/parser.c

```diff
@@ -152,40 +141,22 @@ struct parse_arg_ctx {
 	int hit_usage;
 	enum parse_arg_state state;
 
-	struct uk_libparam_libdesc *ld;
 	struct uk_libparam_param *p;
 };
 
-static struct uk_libparam_libdesc *find_libdesc(const char *libname,
-						__sz libname_len)
-{
-	struct uk_libparam_libdesc *ld_iter;
-
-	UK_LIBPARAM_FOREACH_LIBDESC(ld_iter) {
-		/*
-		 * We want an exact match, because libname might not
-		 * be zero terminated, we need to compare library name and the
-		 * lengths separately.
-		 */
-		if ((strncmp(ld_iter->prefix, libname, libname_len) == 0)
-		    && (ld_iter->prefix[libname_len] == '\0'))
-			return ld_iter;
-	}
-	return NULL;
-}
-
-static struct uk_libparam_param *find_libparam(struct uk_libparam_libdesc *ld,
+static struct uk_libparam_param *find_libparam(const char *libname,
+					       __sz libname_len,
 					       const char *paramname,
 					       __sz paramname_len)
 {
 	struct uk_libparam_param *p;
-	__sz i;
-
-	UK_ASSERT(ld);
 
-	UK_LIBPARAM_FOREACH_PARAMIDX(ld, i) {
-		p = UK_LIBPARAM_PARAM_GET(ld, i);
+	UK_ASSERT(libname);
+	UK_ASSERT(paramname);
 
+	UK_LIBPARAM_FOREACH_PARAM(p) {
```

Comment:
Perhaps a `UK_ASSERT(p);` here as well, since the other FOREACH's have it as well?